lib/ansible/modules/cloud/azure/azure_rm_disk.py

```diff
@@ -0,0 +1,352 @@
+                                            supports_check_mode=True)
+
+    def exec_module(self, **kwargs):
+
+        for key in self.module_arg_spec:
+            setattr(self, key, kwargs[key])
+
+        if self.state == 'absent' and self.managed_by:
+            self.fail('Cannot delete a disk attached to a virtual machine.')
+
+        results = dict()
+        changed = False
+        changeAttached = False
+        disk = None
+
+        if not self.location:
+            # Set default location
+            resource_group = self.get_resource_group(self.resource_group)
+            self.location = resource_group.location
+
+        try:
+            self.log('Fetching disk {0}'.format(self.name))
+            disk = self.compute_client.disks.get(self.resource_group, self.name)
+            self.check_provisioning_state(disk, self.state)
+            results = disk_to_dict(disk)
+
+            # check the update
+            if self.state == 'present':
+                if self.sku:
+                    if results['sku'] != self.sku:
+                        self.log("CHANGED: disk {0} sku".format(self.name))
+                        changed = True
+
+                if self.size_gb:
+                    if results['size_gb'] != self.size_gb:
+                        self.log("CHANGED: disk {0} size".format(self.name))
+                        changed = True
+            elif self.state == 'absent':
+                changed = True            
+        except CloudError:
+            # the disk does not exist
+            if self.state == 'present':
+                changed = True
+
+        # check the attach and detach
+        if self.managed_by != results.get('managed_by'):
+            changeAttached = True
+
+        self.results['changed'] = changed or changeAttached
+        self.results['state'] = results
+
+        if not self.check_mode:
+
+            if self.state == 'present' and changed:
+                if not disk:
+                    # create new disk
+                    self.log('Creating disk {0}'.format(self.name))
+                    # create creation_data TODO
+                    creation_data = CreationData(DiskCreateOption.empty)
+                    params = Disk(
+                        location=self.location,
+                        creation_data=creation_data,
+                        tags=self.tags or {},
+                        sku=DiskSku(self.sku),
+                        disk_size_gb=self.size_gb,
+                    )
+                else:
+                    params = Disk(
+                        location=results['location'],
+                        creation_data=disk.creation_data,
+                        tags=results['tags'] or {},
+                        sku=DiskSku(self.sku or results['sku']),
+                        disk_size_gb=self.size_gb or results['size_gb'],
+                    )
+                disk = self.create_or_update_disk(params)

```

Comment:
You can just pass dictionary here and it will be converted to CreationData automatically by create_or_update_disk()